src/engine/strat_engine/serde_structs.rs

```diff
@@ -12,12 +12,47 @@
 // can convert to or from them when saving our current state, or
 // restoring state from saved metadata.
 
-use serde::Serialize;
+use serde::{Serialize, Serializer};
 
 use devicemapper::{Sectors, ThinDevId};
 
 use crate::engine::types::{DevUuid, FilesystemUuid};
 
+const MAXIMUM_STRING_SIZE: usize = 255;
+
+// Safely returns a truncated String. Can not use truncate() as that panics if
+// len is not on a char boundary.
+fn safe_truncate(value: &str, len: usize) -> String {
+    let bytes_value = value.as_bytes();
+    let len = std::cmp::min(len, bytes_value.len());
+    String::from_utf8_lossy(&bytes_value[..len])
+        .trim_matches(std::char::REPLACEMENT_CHARACTER)
+        .to_string()
```

Comment:
Hmm, why not just index the `String`? That would avoid the conversion to bytes and it could be converted directly from a `str` back to a `String`.

Reply:
The ```str``` index operation is just like ```truncate()```; it'll panic if I try to split a character encoding in the middle. We allow multibyte characters in filesystem names and pool names and it makes sense to allow them in UserInfo property, too.

Reply:
Ah, I read up more on strings in Rust and I think that we'll at the very least want to add in a comment to use [this method](https://doc.rust-lang.org/std/string/struct.String.html#method.floor_char_boundary) once it stabilizes.

Reply:
Comment added!

Reply:
I also just found [this method](`https://doc.rust-lang.org/std/primitive.str.html#method.is_char_boundary`) and I think this would simplify the code quite a bit if we used truncate instead, but just checked it beforehand to see if it's on the byte-boundary and move the index back if it isn't. That would avoid the unnecessary allocation that we have here.

Reply:
@jbaublitz There is a missing link in your comment...

Reply:
@jbaublitz Maybe you're thinking of ```is_char_boundary```?

Reply:
@jbaublitz Yes you were! GitHub hid [this link](https://doc.rust-lang.org/std/primitive.str.html#method.is_char_boundary), though, because the URL was between '`' chars. Sneaky.